ksql-streams/src/main/java/io/confluent/ksql/execution/streams/timestamp/LoggingTimestampExtractor.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.streams.timestamp;
+
+import static io.confluent.ksql.serde.util.SerdeProcessingLogMessageFactory.deserializationErrorMsg;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.confluent.ksql.logging.processing.ProcessingLogger;
+import java.util.Objects;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.streams.processor.TimestampExtractor;
+
+/**
+ * A wrapper around {@code TimestampExtractor} that can be configured to suppress any
+ * errors and instead returns a negative timestamp (indicating to streams that the message
+ * should be ignored). Additionally, this class ensures that any errors are logged to the
+ * processing log (even the fatal ones) for visibility.
+ */
+public class LoggingTimestampExtractor implements TimestampExtractor {
```

Comment:
You might want to implement from `KsqlTimestampExtractor` to we can support this on CSAS/CTAS statements too.

Reply:
I think it's already supported because this just wraps other extractors (we always return this extractor). I'm not sure exactly what you mean here?

Reply:
ah `KsqlTimestampExtractor` is something new. I'll do this